### PR TITLE
feat: [CDS-42131]: added regions field in aws connector and amazon s3 aritfact onprem-hotfix

### PR DESCRIPTION
**Pull request by vikrant-harness**
### Summary

added regions field in aws connector and amazon s3 aritfact onprem-hotfix

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 31 Issues 
* Code issue 31 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10832)

**Comment by vikrant-harness** [comment]
trigger cypress

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/CreateStackRef.tsx [hunk 1]
```diff
@@ -86,6 +86,7 @@ export const CreateStack = (
   const [awsStates, setAwsStates] = useState<SelectOption[]>([])
   const [awsRoles, setAwsRoles] = useState<MultiSelectOption[]>([])
   const [awsRef, setAwsRef] = useState<string>(initialValues?.spec?.configuration?.connectorRef)
+  const [regionsRef, setRegionsRef] = useState<string>(initialValues?.spec?.configuration?.region)
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/CreateStackRef.tsx:89]: This naming seems confusing being state value. It seems its ref value. We can simply use "region" or "refRegions"

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/CreateStackRef.tsx [hunk 2]
```diff
@@ -173,10 +175,18 @@ export const CreateStack = (
       }
       setAwsRoles(roles)
     }
-    if (!roleData && !isEmpty(awsRef) && getMultiTypeFromValue(awsRef) === MultiTypeInputType.FIXED) {
+  }, [roleData, awsRef, regionsRef])
+
+  useEffect(() => {
+    if (
+      !isEmpty(awsRef) &&
+      getMultiTypeFromValue(awsRef) === MultiTypeInputType.FIXED &&
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/CreateStackRef.tsx:183]: You can define util function that will return if MultiTypeInputType is "FIXED"

`const isFixedMultiTypeInputType = (inputType) => inputType === MultiTypeInputType.FIXED`

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/InputSteps/InputStep.tsx [hunk 3]
```diff
@@ -54,6 +60,7 @@ function CreateStackInputStepRef<T extends CreateStackData = CreateStackData>(
   const [selectedCapabilities, setSelectedCapabilities] = useState<MultiSelectOption[]>([])
   const [selectedStackStatus, setSelectedStackStatus] = useState<MultiSelectOption[]>([])
   const [awsRef, setAwsRef] = useState(get(allValues, 'spec.configuration.connectorRef'))
+  const [regionsRef, setRegionsRef] = useState(get(allValues, 'spec.configuration.region'))
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/InputSteps/InputStep.tsx:63]: Same as above

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/InputSteps/InputStep.tsx [hunk 4]
```diff
@@ -216,11 +232,16 @@ function CreateStackInputStepRef<T extends CreateStackData = CreateStackData>(
             disabled={readonly}
             width={300}
             setRefValue
-            onChange={(value: any, _unused, _notUsed) => {
+            onChange={(selected: any, _typeValue, type) => {
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/CreateStack/InputSteps/InputStep.tsx:235]: Can we add the proper type selected here?

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/DeleteStack/DeleteStackInputSteps.tsx [hunk 5]
```diff
@@ -9,30 +9,43 @@ import React, { useState, useEffect } from 'react'
...
 ): React.ReactElement {
-  const { inputSetData, readonly, path, allowableTypes, formik, allValues } = props
+  const { inputSetData, readonly, path, allowableTypes, allValues } = props
   const { getString } = useStrings()
   const { expressions } = useVariablesExpression()
   const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
   const [regions, setRegions] = useState<MultiSelectOption[]>([])
   const [awsRoles, setAwsRoles] = useState<MultiSelectOption[]>([])
   const [awsRef, setAwsRef] = useState<string>('')
+  const [regionsRef, setRegionsRef] = useState(get(allValues, 'spec.configuration.region'))
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/DeleteStack/DeleteStackInputSteps.tsx:48]: same as above

File: src/modules/75-cd/components/PipelineSteps/CloudFormation/DeleteStack/DeleteStackInputSteps.tsx [hunk 6]
```diff
@@ -82,12 +96,19 @@ export function DeleteStackInputStepRef<T extends DeleteStackData = DeleteStackD
...
       setAwsRoles(roles)
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [roleData, roleRequired, awsRef])
 
-    if (!roleData && roleRequired && (awsRef || connectorRef)) {
+  useEffect(() => {
+    if (
+      !isEmpty(awsRef) &&
+      getMultiTypeFromValue(awsRef) === MultiTypeInputType.FIXED &&
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/CloudFormation/DeleteStack/DeleteStackInputSteps.tsx:105]: same as above

File: src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx [hunk 7]
```diff
@@ -69,6 +86,33 @@ const Content = (props: AmazonS3ContentProps): JSX.Element => {
     get(initialValues?.artifacts, `${artifactPath}.spec.filePathRegex`, '')
   )
 
+  const fixedRegionValue = getDefaultQueryParam(
+    artifact?.spec?.region,
+    get(initialValues?.artifacts, `${artifactPath}.spec.region`, '')
+  )
+
+  const [connectorRef, setConnectorRef] = React.useState<string>()
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ArtifactSource/AmazonS3ArtifactSource/AmazonS3ArtifactSource.tsx:94]: Same as above

**Review by chetanvekariya-harness — APPROVED** [review]
In your next PR, please fix **any** types and write proper test cases for region field in AmazonS3.

**Pull request closed without merging**